src/components/ellenTicTacToe.tsx

```diff
@@ -0,0 +1,240 @@
+          {this.renderSquare(5)}
+        </div>
+        <div className="board-row">
+          {this.renderSquare(6)}
+          {this.renderSquare(7)}
+          {this.renderSquare(8)}
+        </div>
+      </div>
+    );
+  }
+}
+interface GameState {
+  history: Array<{ squares: string[]; clickedSquare?: number[] }>;
+  stepNumber: number;
+  xIsNext: boolean;
+}
+// tslint:disable-next-line:max-classes-per-file
+export class Game extends React.Component<{}, GameState> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      history: [
+        {
+          squares: Array(9).fill(null),
+        },
+      ],
+      stepNumber: 0,
+      xIsNext: true,
+    };
+  }
+  handleClick(i: number) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          clickedSquare: [Math.floor((i % 3) + 1), Math.floor(i / 3 + 1)],
+          squares,
+        },
+      ]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext,
+    });
+  }
+  jumpTo(step: number) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0,
+    });
+  }
+  render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
+    const moves = history.map(
+      (step: { squares: string[]; clickedSquare: number[] }, move: number) => {
+        const clickedSquare = step.clickedSquare;
+        const desc = move
+          ? `Go to move #${move} (Col: ${clickedSquare[0]} Row:${
+              clickedSquare[1]
+            })`
+          : 'Go to game start';
+        return (
+          <li key={move}>
+            <button
+              className={css(
+                {
+                  font: 'inherit',
+                  margin: 3,
+                  overflow: 'visible',
+                  textTransform: 'none',
+                },
+                { ':focus': { fontWeight: 'bold' } },
+              )}
+              onClick={() => this.jumpTo(move)}
+            >
+              {desc}
+            </button>
+          </li>
+        );
+      },
+    );
+    let status;
+    if (winner) {
+      status = 'Winner: ' + winner[0];
+    } else if (current.squares.includes(null) === false) {
+      status = 'Tie: No available moves remaining';
+    } else {
+      status = 'Next player: ' + (this.state.xIsNext ? 'X' : 'O');
+    }
+    return (
+      <div
+        className={css({
+          display: 'flex',
+          flexDirection: 'row',
+        })}
+      >
+        <div className="game-board">
+          <Board
+            squares={current.squares}
+            onClick={i => this.handleClick(i)}

```

Comment:
`onClick={this.handleClick}` (you're just passing `i`) through.

You might run into issues with `this` being rebound; Jack and Philippa have definitely both dealt with it, so can help if I'm not about.